lib/common.js

```diff
@@ -474,28 +772,39 @@ function sharkStreams(req, res, next) {
         // We replace the actual size, in case it was streaming, and
         // the content-md5 we actually calculated on the wire
         req._contentMD5 = check.digest('base64');
-        req._size = check.bytes;
+
+        // For AWS chunked MPU, preserve the decoded size we set earlier
+        // Don't let CheckStream override with potentially incorrect size
+        if (req._awsChunkedMPU && req._awsChunkedExpectedSize) {
+            req.log.debug({
+                checkStreamBytes: check.bytes,
+                expectedDecodedSize: req._awsChunkedExpectedSize,
+                preservingExpectedSize: true
+            }, 'S3_MPU: Preserving expected decoded size' +
+               ' for AWS chunked upload');
+            req._size = req._awsChunkedExpectedSize;
+        } else {
+            req._size = check.bytes;
+        }
         barrier.done('client');
     }
 
-    req.once('end', onEnd);
+    // FIX: For AWS chunked uploads, listen to decoder end instead of
```

Comment:
Is this a blocker before integration (FIX)?

Reply:
I missused 'FIX'; I meant I fixed it by adding that line. I'll remove 'FIX' from the comment.